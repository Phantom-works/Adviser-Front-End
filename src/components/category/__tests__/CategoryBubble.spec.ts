import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CategoryBubble from '../CategoryBubble.vue'

describe('CategoryBubble Snapshots', () => {
  // Arrange

  it('Snapshot text empty ', () => {
    //Arrange
    const category = {
      id: 0,
      title: ''
    }
    // Act
    const wrapper = mount(CategoryBubble, {
      props: {
        category: category
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text long ', () => {
    //Arrange
    const category = {
      id: 0,
      title: 'zerdy8y;ljsdrty88;reasdrty;yaetjsdl'
    }
    // Act
    const wrapper = mount(CategoryBubble, {
      props: {
        category: category
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text small ', () => {
    //Arrange
    const category = {
      id: 0,
      title: 'hallo'
    }
    // Act
    const wrapper = mount(CategoryBubble, {
      props: {
        category: category
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
