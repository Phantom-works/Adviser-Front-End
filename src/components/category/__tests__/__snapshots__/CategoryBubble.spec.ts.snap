// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryBubble Snapshots > Snapshot Bubble Default 1`] = `
"<div data-v-577a7027=\\"\\" class=\\"Bubble\\">
  <h3 data-v-577a7027=\\"\\"> <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
</div>"
`;

exports[`CategoryBubble Snapshots > Snapshot text empty  1`] = `
"<div data-v-577a7027=\\"\\" class=\\"Bubble\\">
  <h3 data-v-577a7027=\\"\\"> <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
</div>"
`;

exports[`CategoryBubble Snapshots > Snapshot text long  1`] = `
"<div data-v-577a7027=\\"\\" class=\\"Bubble\\">
  <h3 data-v-577a7027=\\"\\">zerdy8y;ljsdrty88;reasdrty;yaetjsdl <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
</div>"
`;

exports[`CategoryBubble Snapshots > Snapshot text small  1`] = `
"<div data-v-577a7027=\\"\\" class=\\"Bubble\\">
  <h3 data-v-577a7027=\\"\\">hallo <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
</div>"
`;
